swap-orchestrator/src/compose.rs

```diff
@@ -216,6 +242,22 @@ fn build(input: OrchestratorInput) -> String {
         .format("%Y-%m-%d %H:%M:%S UTC")
         .to_string();
 
+    let arti_segment = if input.want_arti {
+        format!(
+            "\
+  arti:
+    container_name: arti
+    {image_arti}
+    restart: unless-stopped
+    expose:
+      - {port_arti_socks}\
```

Comment:
We should explicitly tell arti to listen on this port.

Reply:
We have no control over this: [the container](https://hub.docker.com/r/zydou/arti) listens on fixed ports [with no configuration](https://github.com/zydou/GFW/blob/master/arti/Dockerfile).